packages/expo-notifications/src/Notifications.types.ts

```diff
@@ -183,11 +183,13 @@ export enum AndroidNotificationPriority {
   MAX = 'max',
 }
 
-export type NotificationContent = {
+export type NotificationContent<T = { [key: string]: unknown }> = {
   title: string | null;
   subtitle: string | null;
   body: string | null;
-  data: { [key: string]: unknown };
+  data: T & {
+    [key: string]: unknown;
+  };
```

Comment:
```suggestion
  data: T;
```
This should be enough here, cause you passed the default type for `T`, which is `Record<string, unknow>.

Reply:
I'll submit these changes today!